src-php/Traits/IsAccessible.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Silvanite\NovaToolPermissions\Traits;
+
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Input;
+use Silvanite\NovaToolPermissions\Access;
+
+trait IsAccessible
+{
+    public function initializeIsAccessible()
+    {
+        $this->with = array_merge($this->with, [
+            'access',
+        ]);
+    }
+
+    protected static function boot()
+    {
+        parent::boot();
+        static::created(function ($model) {
+            $model->access()
+                ->create(['roles' => Arr::wrap(Input::get('access_roles', []))]);
+        });
+        static::deleting(function ($model) {
+            if ($model->access) {
+                $access = $model->access;
+                $access->delete();
+            }
+        });
+    }
+
+    public function access()
+    {
+        return $this->morphOne(Access::class, 'accessible');
+    }
+ 
+    public function getAccessRolesAttribute()
+    {
+        return $this->access->roles ?? [];
+    }
+
+    public function setAccessRolesAttribute($roles)
+    {
+        if ($access = $this->access) {
+            $access->roles = $roles;
+            return $access->save();
+        }
```

Comment:
So one edge case I can see is that if you already have data in your table and add this functionality later on, the `created` hook isn't triggering and you will not be able to add any access rules to the model. Could either solve this through the `updating` hook or check here if the model has an ID and then call `access()->create([...]` if it does (e.g. is being updated but doesn't have an access associated.

Reply:
if we check for an ID here, won't this get ran twice when actually updating a model? Should we check for an ID and that the model doesn't currently have an access relation?